specs/trip_dispatch_spec.rb

```diff
@@ -89,4 +90,35 @@
       passenger.trips.must_include trip
     end
   end
+
+  describe "request_trip method" do
+    before do
+      @test_dispatch = RideShare::TripDispatcher.new
+
+      @test_passenger = RideShare::Passenger.new({id: 4, name: "Ervin Wiza", phone: "272-041-9587"})
+
+      @driver = RideShare::Driver.new(id: 8, name: "Shania Olson", vin: "KPH7TNNL14MDUFNF8", status: :UNAVAILABLE)
+    end
+
+    it "raises an error if no id is given" do
+      @test_dispatch = RideShare::TripDispatcher.new
+      proc {
+        @test_dispatch.request_trip()
+      }.must_raise ArgumentError
+    end
+
+    it "only uses a valid user id" do
+      proc{ @test_dispatch.request_trip(0) }.must_raise ArgumentError
+    end
+
+    it "it returns an instance of a trip" do
+      @test_dispatch.request_trip(23).must_be_instance_of RideShare::Trip
+    end
+
+    it "updates Trip Dispatcher with the newest trip" do
+      total_trips = @test_dispatch.trips.length
+      @test_dispatch.request_trip(23)
+      @test_dispatch.trips.length.must_equal total_trips + 1
```

Comment:
Other questions I have:

- Does the trip have the correct `Driver` and `Passenger` listed?
- Were the trip lists on the `Driver` and `Passenger` updated to include the new trip?
- What happens if there are no available drivers?

The first two I'm pretty sure your code does correctly, but the only way to know for sure is to test it!

For the last, this is an important case, and is exactly the sort of situation where software will do something unexpected. You might set up the test as follows:

```ruby
it 'raises an error if no driver is available' do
  @test_dispatch.drivers.each do |driver|
    # This may require an attr_writer in Driver
    driver.status = :UNAVAILABLE
  end
  proc {
    @test_dispatch.request_trip(2)
  }.must_raise StandardError
end
```